java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/job/execution/CompactionTask.java

```diff
@@ -137,14 +142,37 @@ public Instant handleMessages(Instant startTime, Consumer<RecordsProcessedSummar
         return timeSupplier.get();
     }
 
+    private RecordsProcessedSummary compact(CompactionJob job) throws Exception {
+        Instant jobStartTime = timeSupplier.get();
+        LOGGER.info("Compaction job {}: compaction called at {}", job.getId(), jobStartTime);
+        jobStatusStore.jobStarted(job, jobStartTime, taskId);
+        propertiesReloader.reloadIfNeeded();
+        RecordsProcessed recordsProcessed = compactor.compact(job);
+        Instant jobFinishTime = timeSupplier.get();
+        RecordsProcessedSummary summary = new RecordsProcessedSummary(recordsProcessed, jobStartTime, jobFinishTime);
+        jobStatusStore.jobFinished(job, summary, taskId);
+        logMetrics(job, summary);
+        return summary;
+    }
+
+    private void logMetrics(CompactionJob job, RecordsProcessedSummary summary) {
+        LOGGER.info("Compaction job {}: finished at {}", job.getId(), summary.getFinishTime());
+        METRICS_LOGGER.info("Compaction job {}: compaction run time = {}", job.getId(), summary.getDurationInSeconds());
+        METRICS_LOGGER.info("Compaction job {}: compaction read {} records at {} per second", job.getId(),
+                summary.getRecordsRead(), String.format("%.1f", summary.getRecordsReadPerSecond()));
+        METRICS_LOGGER.info("Compaction job {}: compaction wrote {} records at {} per second", job.getId(),
+                summary.getRecordsWritten(), String.format("%.1f", summary.getRecordsWrittenPerSecond()));
+        jobStatusStore.jobFinished(job, summary, taskId);
```

Comment:
This is called twice. From the method name it sounds like this shouldn't be done here?